File: leetcode/100-999/145. Binary Tree Postorder Traversal/solution_4.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution 
{
    public IList<int> PostorderTraversal(TreeNode root) 
    {
        List<int> values = new List<int>();

        if (root == null) return values;

        List<TreeNode> tree = new List<TreeNode>() { root };

        while (tree.Count > 0)
        {
            root = tree[tree.Count - 1];
            tree.RemoveAt(tree.Count - 1);

            values.Insert(0, root.val);

            if (root.left != null)
                tree.Add(root.left);

            if (root.right != null)
                tree.Add(root.right);
        }

        return values;
    }
}

File: leetcode/100-999/145. Binary Tree Postorder Traversal/solution_4.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root) 
    {
        vector<int> values;

        if (!root) return values;

        vector<TreeNode*> tree = {root};

        while (!tree.empty())
        {
            root = tree.back();
            tree.pop_back();

            values.insert(values.begin(), root->val);

            if (root->left)
                tree.push_back(root->left);

            if (root->right)
                tree.push_back(root->right);
        }

        return values;
    }
};